cmd/thanos/query.go

```diff
@@ -73,6 +74,7 @@ func registerQuery(app *extkingpin.App) {
 		Default("20").Int()
 
 	lookbackDelta := cmd.Flag("query.lookback-delta", "The maximum lookback duration for retrieving metrics during expression evaluations. PromQL always evaluates the query for the certain timestamp (query range timestamps are deduced by step). Since scrape intervals might be different, PromQL looks back for given amount of time to get latest sample. If it exceeds the maximum lookback delta it assumes series is stale and returns none (a gap). This is why lookback delta should be set to at least 2 times of the slowest scrape interval. If unset it will use the promql default of 5m.").Duration()
+	dynamicLookbackDelta := cmd.Flag("query.dynamic-lookback-delta", "Allow for larger lookback duration for queries based on resolution.").Hidden().Default("true").Bool()
```

Comment:
Quite busy recently with KubeCon talks, but I was thinking about one more thing we might want to check before deciding on this.

We actually now the best the resolution of the data... when we return it. Store GW depending on options can either return all 5m resolutions, or sometimes full raw resolution or sometimes mix: a bit of raw and a bit of 1h resolution. Wonder if this solution will work for those cases.. and what we actually expect from this? :thinking:

Reply:
> We actually now the best the resolution of the data... when we return it. Store GW depending on options can either return all 5m resolutions, or sometimes full raw resolution or sometimes mix: a bit of raw and a bit of 1h resolution. Wonder if this solution will work for those cases.. and what we actually expect from this? 🤔

Well, result with mixed resolutions was the original case @IKSIN reported. I can foresee one behaviour change there - for range queries with sufficiently large `step` we might (but not necessarily will, dedpending on data and requested window) detect stale timeseries a bit later than we should have done. On the other hand, I doubt anyone will notice it given the requested step should be over 25m / 5h for dynamic lookback interval to even kick in.

Moreover, this adds the possibility to actually set a lower lookback interval for raw queries without sacrificing the downsampled views, allowing for more precise results when it trully maters (low `step` requested).

Reply:
Hm, I think this makes sense - so you say lookback delta can be too large in some cases, but it's a tradeoff with auto-resolution :thinking: 
I think that might make good sense for now (: